xarray/tests/__init__.py

```diff
@@ -114,6 +114,18 @@ def requires_bottleneck(test):
     return test if has_bottleneck else pytest.mark.skip('requires bottleneck')(test)
 
 
+flakey = pytest.mark.skipif(
+    pytest.config.getoption("--skip-flakey"),
+    reason="set --skip-flakey option to skip flakey tests"
+)
+
+
+slow = pytest.mark.skipif(
+    not pytest.config.getoption("--run-slow"),
+    reason="set --run-slow option to run slow tests"
```

Comment:
The h5netcdf many file tests are not flaky: they *always fail*. So let's mark as a "expected failure" with `pytest.mark.xfail` instead.

I would argue that the "CI specific" tests should actually be considered flaky: if they fail on our CI, they are likely to also fail for users and redistributors as well (e.g., consider someone packaging xarray for conda-forge or ubuntu). Since these failures do not indicate a failed installation, we shouldn't run them by default.

I'll push a commit shortly with these changes...